### PR TITLE
add gitsign config subpackage

### PR DESCRIPTION
**Pull request by cpanato**
- add gitsign config subpackage

### CODE REVIEW

**Review by imjasonh — CHANGES_REQUESTED** [review]
I'd like @wlynch's ✅ to make sure this is right.

File: gitsign.yaml [hunk 1]
```diff
@@ -13,6 +13,18 @@ pipeline:
...
+subpackages:
+  - name: "${{package.name}}-config"
+    description: "GitSign config"
+    pipeline:
+      - runs: |
+          mkdir -p ${{targets.subpkgdir}}/etc/
+          git config --global commit.gpgsign true       # Sign all commits
+          git config --global tag.gpgsign true          # Sign all tags
+          git config --global gpg.x509.program gitsign  # Use gitsign for signing
+          git config --global gpg.format x509           # gitsign expects x509 args
```
Comment (wlynch) [inline gitsign.yaml:25]:
```suggestion
          git config --system commit.gpgsign true       # Sign all commits
          git config --system tag.gpgsign true          # Sign all tags
          git config --system gpg.x509.program gitsign  # Use gitsign for signing
          git config --system gpg.format x509           # gitsign expects x509 args
```

I think we probably want `--system` here so it configures for any user. (`--global` is global for the current user)

Comment (cpanato) [reply gitsign.yaml]: done

**Review by wlynch — COMMENTED** [review]
LGTM (can't approve since I'm not a wolfi-dev member 😢 )

**Review by imjasonh — APPROVED** [review]

**Pull request merged**